### ISSUE TITLE
Error instantiating bean of type  [io.micronaut.security.token.jwt.validator.JwtTokenValidator]

### ISSUE THREAD

**Issue by mircoporetti** [micronaut-projects/micronaut-security#1036]
### Issue description

As discussed with @sdelamo we have the following issue:

On Kubernetes we have a Deployment of a Micronaut app. Sometimes it happens that after deploying, the pod continues to restart and it gives us some log:

```
13:07:51.152 [default-nioEventLoopGroup-1-3] WARN  i.n.c.AbstractChannelHandlerContext - An exception 'io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.micronaut.security.token.jwt.validator.JwtTokenValidator]13:07:51.152 [default-nioEventLoopGroup-1-3] WARN  i.n.c.AbstractChannelHandlerContext - An exception 'io.micronaut.context.exceptions.BeanInstantiationException: Error instantiating bean of type  [io.micronaut.security.token.jwt.validator.JwtTokenValidator]
Message: Read TimeoutPath Taken: new SecurityFilter(Collection securityRules,Collection authenticationFetchers,SecurityConfiguration securityConfiguration) --> new SecurityFilter(Collection securityRules,[Collection authenticationFetchers],SecurityConfiguration securityConfiguration) --> new TokenAuthenticationFetcher([Collection tokenValidators],TokenResolver tokenResolver,ApplicationEventPublisher eventPublisher) --> new JwtTokenValidator([Collection signatureConfigurations],Collection encryptionConfigurations,Collection genericJwtClaimsValidators,JwtAuthenticationFactory jwtAuthenticationFactory)' 
```

After the next deploy, it usually runs normally.
It doesn't happen  running the application locally.

We use the following configuration for security:

```
security:
    authentication: idtoken
    enabled: true
    oauth2:
      clients:
        okta:
          openid:
            issuer: ${OIDC_ISSUER_DOMAIN: ......
    endpoints:
      logout:
      get-allowed: true
    intercept-url-map:
      - pattern: /**
        access:
          - isAnonymous()
```
          
We use isAnonymous() as default access and then we annotate only the endpoints that have to be authorized, using `@Secured(SecurityRule.IS_AUTHENTICATED)`

Thanks

**Comment by sdelamo** [comment]
I am not sure but I think the issue maybe that `JWKSSet.load` is timing out when trying to fetch the JWKS fom Okta Authorization server. 

https://github.com/micronaut-projects/micronaut-security/blob/master/security-jwt/src/main/java/io/micronaut/security/token/jwt/signature/jwks/JwksSignature.java#L#L216

Do you think this maybe when trying to hit the health endpoint to see if the pod is healthy?

**Comment by mircoporetti** [comment]
That's an idea. But I'm wondering why the pod continues to restart with the same error and after deploying again usually it works without the problem

**Comment by jeremyg484** [comment]
This issue looks to be the same as #1633 which I believe we have now fixed with #1641. I am closing with that assumption. If it proves to not be fixed by #1641 then we can re-open and investigate further.
